suprb/optimizer/rule/mutation.py

```diff
@@ -69,47 +87,108 @@ class Normal(RuleMutation):
     def ordered_bound(self, rule: Rule, random_state: RandomState):
         # code inspection gives you a warning here but it is ineffectual
         rule.match.bounds += random_state.normal(scale=self.sigma,
-                                        size=rule.match.bounds.shape)
+                                                 size=rule.match.bounds.shape)
         rule.match.bounds = np.sort(rule.match.bounds, axis=1)
 
+    def unordered_bound(self, rule: Rule, random_state: RandomState):
+        rule.match.bounds += random_state.normal(scale=self.sigma,
+                                                 size=rule.match.bounds.shape)
+
+    def centre_spread(self, rule: Rule, random_state: RandomState):
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        bounds = rule.match.bounds
+        bounds[:, 0] += random_state.normal(scale=self.sigma[0],
+                                            size=rule.match.bounds.shape[0])
+        bounds[:, 1] += random_state.normal(scale=self.sigma[1],
+                                            size=rule.match.bounds.shape[0])
+
+    def min_percentage(self, rule: Rule, random_state: RandomState):
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        bounds = rule.match.bounds
+        bounds[:, 0] += random_state.normal(scale=self.sigma[0],
+                                            size=rule.match.bounds.shape[0])
+        bounds[:, 1] += random_state.normal(scale=self.sigma[1],
+                                            size=rule.match.bounds.shape[0])
+
 
 class Halfnorm(RuleMutation):
     """Sample with (half)normal distribution around the center."""
 
-    def mutation(self, dimensions: int, random_state: RandomState):
-        return halfnorm.rvs(scale=self.sigma / 2, size=dimensions,
-                            random_state=random_state)
-
     def ordered_bound(self, rule: Rule, random_state: RandomState):
         bounds = rule.match.bounds
         mean = np.mean(bounds, axis=1)
-        bounds[:, 0] = mean - self.mutation(dimensions=bounds.shape[0], random_state=random_state)
-        bounds[:, 1] = mean + self.mutation(dimensions=bounds.shape[0], random_state=random_state)
+        bounds[:, 0] = mean - \
+            halfnorm.rvs(scale=self.sigma / 2, size=bounds.shape[0], random_state=random_state)
+
+        bounds[:, 1] = mean + \
+            halfnorm.rvs(scale=self.sigma / 2, size=bounds.shape[0], random_state=random_state)
         rule.match.bounds = np.sort(rule.match.bounds, axis=1)
 
+    def unordered_bound(self, rule: Rule, random_state: RandomState):
+        bounds = rule.match.bounds
+        mean = np.mean(bounds, axis=1)
+        bounds[:, 0] = mean - \
+            halfnorm.rvs(scale=self.sigma / 2, size=bounds.shape[0], random_state=random_state)
+
+        bounds[:, 1] = mean + \
+            halfnorm.rvs(scale=self.sigma / 2, size=bounds.shape[0], random_state=random_state)
```

Comment:
Same as above: `ordered_bounds` can be rewritten by using `unordered_bounds` and then sort it, to get rid of the code duplication